packages/block-editor/src/components/use-block-drop-zone/index.js

```diff
@@ -172,9 +215,15 @@ export default function useBlockDropZone( {
 	const { showInsertionPoint, hideInsertionPoint } =
 		useDispatch( blockEditorStore );
 
-	const onBlockDrop = useOnBlockDrop( targetRootClientId, dropTarget.index, {
-		operation: dropTarget.operation,
-	} );
+	const onBlockDrop = useOnBlockDrop(
+		dropTarget.operation === 'before' || dropTarget.operation === 'after'
+			? parentBlockClientId
+			: targetRootClientId,
```

Comment:
> So this works, but I can't quite figure out how 😅 because at root level, targetRootClientId should be undefined.

I think the comment for this hook is a bit misleading, unfortunately! At the root of the document, `targetRootClientId` will be undefined / an empty string, however for a container block at the root of the document that calls `useBlockDropZone`, it'll pass its `clientId` in as `rootClientId` here: https://github.com/WordPress/gutenberg/blob/7e6177104c1f905ea3adbdd8f94598c5d680975a/packages/block-editor/src/components/inner-blocks/index.js#L218

What this means, is that for a container block, by the time you're inside `useBlockDropZone`, `targetRootClientId` refers to the client id of the container block itself. So for our purposes, we need to pass in `parentBlockClientId` (the parent of the container's clientId) in order for it to work properly. In practice, for a Group block at the root of the document, `targetRootClientId` will be the client id of the Group block, and `parentBlockClientId` will be an empty string, as we're at the root of the document.

Hope that helps, I know it's quite a mouthful! If / when this PR looks ready to land, I'll see if I can write this up as a comment prior to this line so that it's clearer — and / or so that we can remember what we did here 😄

---

> Also I tried logging the operation here while dragging and it always switches to insert before the drop happens.

Oh strange! I didn't notice that, but the drop should be happening with the previously returned `onBlockDrop` that had the last state with the "before" or "after" operation. When you perform a drop, I think it'll immediately clear out the `dropTarget` state back to its default which is `insert`, so in practice while logging we might be observing the returning to the previous state 🤔

I'm possibly either over thinking that part (or slightly confused about it), but in practice, `before`, `after`, and `insert` should all do the same thing, as the only operation that does anything differently in `useOnBlockDrop` is `replace`.

If it's causing any issues though, I can follow-up next week!